src/main/scala/net/snowflake/spark/snowflake/Utils.scala

```diff
@@ -412,18 +409,23 @@ object Utils {
 
   def runQuery(params: Map[String, String], query: String): ResultSet = {
     val conn = getJDBCConnection(params)
-    val rs = conn.createStatement().executeQuery(query)
+    val resultSerializables =
+      conn.createStatement()
+        .executeQuery(query)
+        .asInstanceOf[SnowflakeResultSet]
+        .getResultSetSerializables(Long.MaxValue)
     conn.close()
-    rs
+    if (resultSerializables.size() == 1) {
+      resultSerializables.get(0).getResultSet
+    } else {
+      throw new Exception("Fail to execute query by Utils.runQuery()")
+    }
```

Comment:
Only support small result query？

Reply:
No. It supports all cases. "Long.MaxValue" is passed as expected partition size. It's sure only one object is returned.
Looks it is confusing, I will remove the THROW.